Core/Configuration/RegistryConfiguration.cs

```diff
@@ -3,30 +3,25 @@
 using System.Linq;
 using System.IO;
 using Microsoft.Win32;
+using log4net;
+using Newtonsoft.Json;
 
-namespace CKAN
+namespace CKAN.Configuration
 {
-    public interface IWin32Registry
+    // DEPRECATED: We now use a JSON configuration file. This still exists to facillitate migration.
+    //
+    // N.B., you can resume using this version by changing the instance created in ServiceLocator.
+    public class RegistryConfiguration : IConfiguration
     {
-        string AutoStartInstance { get; set; }
-        void SetRegistryToInstances(SortedList<string, KSP> instances);
-        IEnumerable<Tuple<string, string>> GetInstances();
-        string GetKSPBuilds();
-        void SetKSPBuilds(string buildMap);
-        string DownloadCacheDir { get; set; }
-        long? CacheSizeLimit { get; set; }
-        int RefreshRate { get; set; }
-    }
+        private static readonly ILog Log = LogManager.GetLogger(typeof(RegistryConfiguration));
 
-    public class Win32Registry : IWin32Registry
-    {
         private const           string CKAN_KEY           = @"HKEY_CURRENT_USER\Software\CKAN";
         private static readonly string CKAN_KEY_NO_PREFIX = StripPrefixKey(CKAN_KEY);
 
         private const           string authTokenKey         = CKAN_KEY + @"\AuthTokens";
         private static readonly string authTokenKeyNoPrefix = StripPrefixKey(authTokenKey);
 
-        static Win32Registry()
+        static RegistryConfiguration()
         {
             ConstructKey(CKAN_KEY_NO_PREFIX);
```

Comment:
If a user installs CKAN for the very first time with no JSON file **and** nothing in the registry, I think this will end up initializing the registry key with no data or default data, never to be used, since the `Migrate` logic relies on constructing a `RegistryConfiguration` object. Can we add a way to check whether the key exists without doing that?

Reply:
Good call. I completely forgot to account for that. As it's currently written, the code works, but that would certainly be more correct.

Reply:
How do you feel about deleting the registry keys after migration? I had a brief discussion with @DasSkelett about it earlier, and so right now the code for that is written but not pushed to this branch.

Reply:
> How do you feel about deleting the registry keys after migration?

It's appealing from an architectural cleanliness perspective, but there are also advantages to leaving the old data in place in case something goes wrong, and until the new code establishes a track record of stability, I'd vote for not deleting.

Reply:
On another thought, @HebaruSan is probably right, we should keep it for now, to ease testing and switching around. And since deleting the old registry entries will definitely break backwards-compatibility, it makes sense to release it only in a new major release.